gateway/host_checker_manager.go

```diff
@@ -307,9 +307,17 @@ func (hc *HostCheckerManager) HostDown(urlStr string) bool {
 	log.WithFields(logrus.Fields{
 		"prefix": "host-check-mgr",
 	}).Debug("Key is: ", PoolerHostSentinelKeyPrefix+u.Host)
-	_, ok := hc.unhealthyHostList.Load(PoolerHostSentinelKeyPrefix + u.Host)
+
+	key := PoolerHostSentinelKeyPrefix + u.Host
+	// If the node doesn't perform any uptime checks, query the storage:
+	if !hc.pollerStarted {
```

Comment:
I think you were trying to check if we are polling or not

```suggestion
	if !hc.AmIPolling() {
```

Reply:
@gernest if the gateway is polling (`hc.AmIPolling() == true`), the `hc.pollerStarted` is going to be true. I think this way, we're going to save a lot of the Redis get's that are done in `hc.AmIPolling()`. 
Is there any situation that I might be missing?

Reply:
@tbuchaillot that makes sense.  But did you know `hc.store` can be nil? We also need to to have a nil check before calling it to avoid panics.

```suggestion
	if !hc.pollerStarted && hc.store != nil {
```

Reply:
@gernest Totally agree with that. We do that on `nextTarget` at `reverse_proxy.go` but a second check won't hurt 👍